### PR TITLE
Update futures

### PR DESCRIPTION
**Pull request by Tim-Zhang**
use futures@0.3 instead of futures@0.1
    use tokio@0.3 instead of tokio@0.1
    ...

This pull request should be merged after https://github.com/fsyncd/vsock-rs/pull/2 because the dependency vsock is point to https://github.com/fsyncd/vsock-rs and depend on the pull request 2.

### CODE REVIEW

**Comment by Tim-Zhang** [comment]
Hi @dpeckett , 
I would like to ask for a review
Thank you very much.

**Comment by dpeckett** [comment]
Thanks Tim for the contribution, I'll review the changes and get back to you once complete.

**Comment by Tim-Zhang** [comment]
> Thanks Tim for the contribution, I'll review the changes and get back to you once complete.

Thank you and I have just updated the test.

File: Cargo.toml [hunk 1]
```diff
@@ -10,13 +10,15 @@ exclude = ["test_fixture"]
...
-libc = "0.2.65"
-mio = "0.6.19"
-nix = "0.15.0"
-tokio = "0.1.22"
-vsock = "0.1.4"
+libc = "0.2.71"
+mio = "0.6.20"
+nix = "0.17.0"
+tokio = { version = "0.2", features = ["io-driver", "stream", "macros", "rt-threaded", ] }
+vsock = { git = "https://github.com/Tim-Zhang/vsock-rs", branch = "master" }
```
Comment (Tim-Zhang) [inline Cargo.toml:19]: It should be update after https://github.com/fsyncd/vsock-rs/pull/2 merged.

File: Cargo.toml [hunk 2]
```diff
@@ -10,13 +10,15 @@ exclude = ["test_fixture"]
...
-libc = "0.2.65"
-mio = "0.6.19"
-nix = "0.15.0"
-tokio = "0.1.22"
-vsock = "0.1.4"
+libc = "0.2.71"
+mio = "0.6.20"
+nix = "0.17.0"
+tokio = { version = "0.2", features = ["io-driver", "stream", "macros", "rt-threaded", ] }
+vsock = { git = "https://github.com/fsyncd/vsock-rs", branch = "master" }
```
Comment (Tim-Zhang) [inline Cargo.toml:19]: This should be updated after `vsock-rs` published on crates.io.

Comment (dpeckett) [reply Cargo.toml:19]: I've published the updated crate for `vsock-rs` so can be changed now.

Comment (Tim-Zhang) [reply Cargo.toml]: > I've published the updated crate for `vsock-rs` so can be changed now.

done.

**Comment by dpeckett** [comment]
This should now be published on crates.io, v0.2.1 includes the CLOEXEC changes from vsock-rs. Thanks for your great submission and particularly the effort placed into the tests :).

**Pull request merged**